src/register/RegistrationPage.jsx

```diff
@@ -543,6 +556,44 @@ class RegistrationPage extends React.Component {
       }
     }
 
+    const formFields = Object.keys(this.props.fieldDescriptions).map((fieldName) => {
+      const fieldData = this.props.fieldDescriptions[fieldName];
+      if (fieldData.name === 'country' && fieldData.type === 'select') {
+        return (
+          <CountryDropdown
+            name="country"
+            floatingLabel={intl.formatMessage(messages['registration.country.label'])}
+            options={getCountryList(getLocale())}
+            valueKey="code"
+            displayValueKey="name"
+            value={this.state.country}
+            handleBlur={this.handleOnBlur}
+            handleFocus={this.handleOnFocus}
+            errorMessage={intl.formatMessage(messages['empty.country.field.error'])}
+            handleChange={(value) => this.setState({ country: value })}
+            errorCode={this.state.errorCode}
+            readOnly={this.state.readOnly}
+          />
+        );
+      }
+      if (fieldData.name === 'Honor Code') {
+        return (<HonorCode fieldData={fieldData} value={this.state.values[fieldData.name]} onChangeHandler={this.onChangeHandler} />);
```

Comment:
nit: If the field is on the same line we don't need to add extra brackets
```suggestion
        return <HonorCode fieldData={fieldData} value={this.state.values[fieldData.name]} onChangeHandler={this.onChangeHandler} />;
```